packages/grafana-data/src/types/pluginExtensions.ts

```diff
@@ -340,3 +345,84 @@ type Dashboard = {
   title: string;
   tags: string[];
 };
+
+// Dynamic Command Palette Types
+// --------------------------------------------------------
+
+/**
+ * A single dynamic result item returned by a command palette search provider
+ */
+export type CommandPaletteDynamicResult = {
+  /** Unique identifier for this result (scoped to plugin) */
+  id: string;
+  /** Display title */
+  title: string;
+  /** Optional subtitle or description */
+  description?: string;
+  /** Optional URL to navigate to (alternative to onSelect) */
+  path?: string;
+  /** Optional keywords for better search matching */
+  keywords?: string[];
+  /** Optional section/category override (defaults to plugin category) */
+  section?: string;
+  /** Optional custom data to pass through to the action handler */
+  data?: Record<string, unknown>;
+};
+
+/**
+ * Action handler for when a dynamic result is selected
+ */
+export type CommandPaletteDynamicResultAction = (
+  result: CommandPaletteDynamicResult,
+  helpers: PluginExtensionEventHelpers<PluginExtensionCommandPaletteContext>
+) => void | Promise<void>;
+
+/**
+ * Search provider function that fetches dynamic results
+ */
+export type CommandPaletteDynamicSearchProvider = (
+  context: PluginExtensionCommandPaletteContext
+) => Promise<CommandPaletteDynamicResult[]>;
+
+/**
+ * Configuration for registering a dynamic command palette provider
+ */
+export type PluginExtensionCommandPaletteDynamicConfig = {
+  /** Display name for this search provider */
+  title: string;
+
+  /**
+   * Category/section name for grouping results
+   * @default Plugin ID
+   */
+  category?: string;
```

Comment:
Should this default to plugin ID? I mean I assume that would look a bit ugly, so maybe we can just keep the group blank? Or do apps have some better user readable titles we could use?